src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,50 +1,89 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+const initialMovieState: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState<Movie>(initialMovieState);
+
+  const handleChange = (field: keyof Movie, value: string) => {
+    setMovie((prevMovie) => ({ ...prevMovie, [field]: value }));
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    onAdd(movie);
+
+    setCount((prevCount) => prevCount + 1);
+    setMovie(initialMovieState);
+  };
+
+  const isFormValid
+    = movie.title.trim() !== ''
+    && movie.imgUrl.trim() !== ''
+    && movie.imdbUrl.trim() !== ''
+    && movie.imdbId.trim() !== '';
 
   return (
-    <form className="NewMovie" key={count}>
+    <form onSubmit={handleSubmit} className="NewMovie" key={count}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={movie.title}
+        onChange={(value) => handleChange('title', value)}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={movie.description}
+        onChange={(value) => handleChange('description', value)}
```

Comment:
no need to hardcode each field.
just pass reference on handleChange to TextField component
```suggestion
        onChange={handleChange}
```
also in TextFiled component change type of onChange prop
```
interface Props {
...
onChange: (event: React.ChangeEvent<HtmlInputElement>) => void;
}
```
and to input pass name and onChange reference
```
<input
  name={name}
  onChange={onChange}
  ...
 >
 ```